comp/core/gui/guiimpl/gui.go

```diff
@@ -56,74 +79,111 @@ type Payload struct {
 	CaseID string `json:"caseID"`
 }
 
-// StopGUIServer closes the connection to the HTTP server & removes the authentication token file we created
-func StopGUIServer() {
-	if listener != nil {
-		listener.Close()
-	}
+type dependencies struct {
+	fx.In
+
+	Log       log.Component
+	Config    config.Component
+	Flare     flare.Component
+	Status    status.Component
+	Collector collector.Component
+	Ac        autodiscovery.Component
+	Lc        fx.Lifecycle
 }
 
-// StartGUIServer creates the router, starts the HTTP server & generates the authentication token for access
-func StartGUIServer(port string,
-	flare flare.Component,
-	statusComponent status.Component,
-	collector collector.Component,
-	ac autodiscovery.Component) error {
-	// Set start time...
-	startTimestamp = time.Now().Unix()
+// GUI component implementation constructor
+// @param deps dependencies needed to construct the gui, bundled in a struct
+// @return an optional, depending of "GUI_port" configuration value
+func newGui(deps dependencies) (optional.Option[guicomp.Component], error) {
+
+	guiPort := deps.Config.GetString("GUI_port")
+
+	if guiPort == "-1" {
+		deps.Log.Infof("GUI server port -1 specified: not starting the GUI.")
+		return optional.NewNoneOption[guicomp.Component](), nil
+	}
+
+	g := gui{
+		port:      guiPort,
+		logger:    deps.Log,
+		config:    deps.Config,
+		flare:     deps.Flare,
+		status:    deps.Status,
+		collector: deps.Collector,
+		ac:        deps.Ac,
+	}
 
 	// Instantiate the gorilla/mux router
-	router := mux.NewRouter()
+	g.router = mux.NewRouter()
 
 	// Serve the only public file at the authentication endpoint
-	router.HandleFunc("/authenticate", generateAuthEndpoint)
+	g.router.HandleFunc("/authenticate", g.generateAuthEndpoint)
 
 	// Serve the (secured) index page on the default endpoint
-	router.Handle("/", authorizeAccess(http.HandlerFunc(generateIndex)))
+	g.router.Handle("/", g.authorizeAccess(http.HandlerFunc(generateIndex)))
 
 	// Mount our (secured) filesystem at the view/{path} route
-	router.PathPrefix("/view/").Handler(http.StripPrefix("/view/", authorizeAccess(http.HandlerFunc(serveAssets))))
+	g.router.PathPrefix("/view/").Handler(http.StripPrefix("/view/", g.authorizeAccess(http.HandlerFunc(serveAssets))))
 
 	// Set up handlers for the API
 	agentRouter := mux.NewRouter().PathPrefix("/agent").Subrouter().StrictSlash(true)
-	agentHandler(agentRouter, flare, statusComponent)
+	g.agentHandler(agentRouter)
 	checkRouter := mux.NewRouter().PathPrefix("/checks").Subrouter().StrictSlash(true)
-	checkHandler(checkRouter, collector, ac)
+	g.checkHandler(checkRouter)
```

Comment:
Nitpick: 
Rather than adding this private function to `gui` What about passing the require dependencies to the functions? 

```go
agentHandler(agentRouter, gui.status, gui.log, ....)
```

That way we do not hide the depencies that each function needs.